microsoft/testsuites/hpc/infinibandsuit.py

```diff
@@ -100,3 +103,58 @@ def verify_hpc_over_nd(self, log: Logger, node: Node) -> None:
             assert_that(modprobe.is_module_loaded(module)).described_as(
                 f"Module {module} is not loaded."
             ).is_true()
+
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. Identify the infiniband devices and their cooresponding network interfaces
+        2. Run several ping-pong tests to check RDMA / Infiniband functionality
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            supported_features=[Infiniband],
+            min_count=2,
+        ),
+    )
+    def verify_ping_pong(self, environment: Environment, log: Logger) -> None:
+        # Constants
+        ping_pong_tests = ["ibv_rc_pingpong", "ibv_uc_pingpong", "ibv_ud_pingpong"]
+
+        server_node = environment.nodes[0]
+        client_node = environment.nodes[1]
+
+        # Ensure RDMA is setup
+        run_in_parallel(
+            [
+                partial(client_node.features[Infiniband].initialize),
```

Comment:
If there is no argument for the methods, it doesn't need to use `partial`. The partial is used to wrap with arguments.

Reply:
This doesn't seem to actually run the setup in parallel. I believe that the initialization is called when you access the feature from `node.features`. So how do I pass it a callable object?

Reply:
I used lambda functions. I hope this is a better solution.

Reply:
Cool, I should create a get method to features like tools.